src/overlays/actors/ovl_Boss_03/z_boss_03.c

```diff
@@ -41,157 +269,2321 @@ const ActorInit Boss_03_InitVars = {
+    Animation_MorphToLoop(&this->skelAnime, &gGyorgFastSwimmingAnim, -15.0f);
+    this->workTimer[WORK_TIMER_CURRENT_ACTION] = 100;
+    this->unk_2C4 = 0.0f;
+    this->unk_2B8 = 0.0f;
+    this->unk_242 = arg2;
+}
+
+void Boss03_CatchPlayer(Boss03* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    f32 xDiff;
+    f32 yDiff;
+    f32 zDiff;
+    Actor* new_var;
+
+    this->skelAnime.playSpeed = 2.0f;
+    this->unk_27C = 2.0f;
+    this->unk_276 = 0x1000;
+    this->unk_2BD = true;
+    this->unk_278 = 15.0f;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+
+    SkelAnime_Update(&this->skelAnime);
+
+    xDiff = player->actor.world.pos.x - this->actor.world.pos.x;
+    yDiff = player->actor.world.pos.y - this->actor.world.pos.y;
+    zDiff = player->actor.world.pos.z - this->actor.world.pos.z;
+
+    Math_ApproachS(&this->actor.world.rot.x, Math_FAtan2F(sqrtf(SQ(xDiff) + SQ(zDiff)), -yDiff), 0xA, this->unk_274);
+    Math_ApproachS(&this->bodyYRot,
+                   Math_SmoothStepToS(&this->actor.world.rot.y, Math_FAtan2F(zDiff, xDiff), 0xA, this->unk_274, 0) *
+                       -0.5f,
+                   5, 0x100);
+    Math_ApproachS(&this->unk_274, this->unk_276, 1, 0x100);
+    Math_ApproachF(&this->actor.speedXZ, this->unk_278, 1.0f, this->unk_27C);
+    Math_ApproachF(&this->unk_260, __sinf(this->skelAnime.curFrame * (M_PI / 5.0f)) * 10.0f * 0.01f, 0.5f, 1.0f);
+    Actor_MoveWithoutGravityReverse(&this->actor);
+    Math_ApproachS(&this->actor.shape.rot.x, this->actor.world.rot.x, 2, this->unk_274 * 2);
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, this->unk_274 * 2);
+
+    // If either (Player is on the floor && Player is above water) or (WORK_TIMER_CURRENT_ACTION timer runs out) -> Stop
+    // trying to catch Player
+    if (((player->actor.bgCheckFlags & 1) && (player->actor.shape.feetPos[FOOT_LEFT].y >= WATER_HEIGHT + 8.0f)) ||
+        (this->workTimer[WORK_TIMER_CURRENT_ACTION] == 0)) {
+        if (&this->actor == player->actor.parent) {
+            player->unk_AE8 = 101;
+            player->actor.parent = NULL;
+            player->csMode = 0;
+            func_80165690();
+        }
+
+        func_809E344C(this, globalCtx);
+        // WORK_TIMER_UNK1_A wasn't set
+    } else {
+        f32 phi_f0;
+
+        Math_ApproachF(&player->actor.world.pos.x, this->insideJawPos.x, 1.0f, this->unk_2B8);
+        Math_ApproachF(&player->actor.world.pos.y, this->insideJawPos.y, 1.0f, this->unk_2B8);
+        Math_ApproachF(&player->actor.world.pos.z, this->insideJawPos.z, 1.0f, this->unk_2B8);
+
+        if (this->unk_242 != 0) {
+            phi_f0 = 10.0f;
+        } else {
+            phi_f0 = 2.0f;
+        }
+
+        Math_ApproachF(&this->unk_2B8, 100.0f, 1.0f, phi_f0);
+
+        if (this->unk_2B8 > 30.0f) {
+            // fake match
+            if (((new_var = &this->actor) != player->actor.parent) && (globalCtx->grabPlayer(globalCtx, player) != 0)) {
```

Comment:
Was curious, and it seems the `new_var = &this->actor` isn't needed. Though `new_var` is still needed as padding.